### DISCUSSION TITLE
How to setup to W.E.D to move words like Helix

### DISCUSSION THREAD

**Discussion by jellydn** [chrisgrieser/nvim-spider#47] [Ideas]
Hi @chrisgrieser 

Thank you for your awesome plugin. I wonder if we could have a similar `helix` motion for WORDS. Here is the record from helix.

[![helix](https://i.gyazo.com/1e213faf478d573cdfbdfbbbef1ddd68.gif)](https://gyazo.com/1e213faf478d573cdfbdfbbbef1ddd68)

I'm thinking we could have something on config. Any thoughts on this? T

```lua
return {
  -- Use the w, e, b motions like a spider. Move by subwords and skip insignificant punctuation.
  -- This is a one-of-a-kind "modal" text editor
  {
    "chrisgrieser/nvim-spider",
    lazy = true,
    vscode = true,
    keys = {
      {
        "w",
        "<cmd>lua require('spider').motion('w')<CR>",
        mode = { "n", "o", "x" },
        desc = "spider-w",
      },
      {
        "e",
        "<cmd>lua require('spider').motion('e')<CR>",
        mode = { "n", "o", "x" },
        desc = "spider-e",
      },
      {
        "b",
        "<cmd>lua require('spider').motion('b')<CR>",
        mode = { "n", "o", "x" },
        desc = "spider-b",
      },
      {
        "ge",
        "<cmd>lua require('spider').motion('ge')<CR>",
        mode = { "n", "o", "x" },
        desc = "spider-ge",
      },
    },
    config = function()
      -- TODO: Use uppercase W, E, B to move by WORDs like Helix.
    end,
  },
}
``

**Comment by chrisgrieser** [comment]
You do not need spider for that, the builtin motions W E and B already move by WORD.

**Reply by jellydn** [reply]
Thank you for your quick reply. I guess this screenshot will explain it better my record. @chrisgrieser 

<img width="847" alt="image" src="https://github.com/user-attachments/assets/588fea93-c35f-4c45-8a78-4a4d74bfa8d4">

**Reply by jellydn** [reply]
If that's NOT clear, please let me know and I will record again 2 different motions between `w` and `W` in helix.

**Reply by chrisgrieser** [reply]
I repeat: the WORD motions described in that screenshot are already available as builtin motions in nvim. 

If you want sth else, you are a) not conveying that properly and b) you can achieve that via custom lua patterns, as is described in the Readme.

**Reply by jellydn** [reply]
> already available as builtin motions in nvim
That sounds to me like I didn't explain clearly to you :). It's not built-in.

> you can achieve that via custom lua patterns

So yes, I think that's what I would try. Thank you very much.
